packages/headless/src/state/state-sections.ts

```diff
@@ -374,3 +378,17 @@ export interface InsightInterfaceSection {
    */
   insightInterface: InsightInterfaceState;
 }
+
+export interface InsightCaseContextSection {
+  /**
+   * The insight case context
+   */
+  insightCaseContext: InsightCaseContextState;
+}
+
+export interface InsightSearchSection {
+  /**
+   * The insight search
```

Comment:
I think we can add a bit more detail.
Copying from the SearchState description:
```suggestion
   * The information related to the insight search endpoint.
```